File: crates/swiftlet/examples/gr_rule.rs
use std::sync::Arc;
use swiftlet::{ParserConfig, Swiftlet};

fn main() {
    let grammar = r#"
    start: hello world
    hello: "hello"
    world: "world"
    "#;

    let text = "helloworld";

    let conf = Arc::new(ParserConfig::default());
    let parser = Swiftlet::from_str(grammar)
        .map(|grammar| grammar.parser(conf))
        .expect("failed to build parser");
    match parser.parse(&text) {
        Ok(ast) => {
            ast.print()
            // Output: Tree("start", [Tree("hello", ["hello"]), Tree("world", ["world"])])
        }
        Err(err) => eprintln!("{}", err),
    }
}

File: pyswiftlet/examples/gr_rule.py
"""Grammar rule example using the Swiftlet Python bindings."""

from swiftlet import Swiftlet


GRAMMAR = r"""
start: hello world
hello: "hello"
world: "world"
"""


def main() -> None:
    parser = Swiftlet(GRAMMAR)
    tree = parser.parse("helloworld")
    tree.pretty_print()


if __name__ == "__main__":
    main()
